Lib/asyncio/unix_events.py

```diff
@@ -1373,12 +1373,16 @@ def is_active(self):
     def close(self):
         self._join_threads()
 
-    def _join_threads(self):
+    def _join_threads(self, timeout=None):
         """Internal: Join all non-daemon threads"""
         threads = [thread for thread in list(self._threads.values())
                    if thread.is_alive() and not thread.daemon]
         for thread in threads:
-            thread.join()
+            thread.join(timeout)
+
+        # Clear references to terminated threads
+        self.threads = [thread for thread in list(self._threads.values())
+                        if thread.is_alive() and not thread.daemon]
```

Comment:
Why are you creating a new instance variable `self.threads` here? Did you intend to clean out completed threads from the dict `self._threads`?

Reply:
I used the list comprehension above as an example, for me it looks easier to remove completed threads rather than having a regular list and removing items from the list.

The purpose of this list is to remove completed threads to save memory and to fix the dangling threads issue.

Reply:
Do you mean that you prefer to not create a new list?

Reply:
I modified my PR to use ``self.threads[:] = ...``, so the self.threads list object is not replacd.

Reply:
> Do you mean that you prefer to not create a new list?

No I meant that the tests fail because `self._threads` is a dict, not a list.